GetUserDataProcessingActivity/handler.ts

```diff
@@ -116,48 +118,50 @@ export const createSetUserDataProcessingStatusActivityHandler = (
   // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 ) => (context: Context, input: unknown) =>
   // the actual handler
-  fromEither(ActivityInput.decode(input))
-    .mapLeft<ActivityResultFailure>((reason: t.Errors) =>
+  pipe(
+    input,
+    ActivityInput.decode,
+    E.mapLeft(reason =>
       ActivityResultInvalidInputFailure.encode({
         kind: "INVALID_INPUT_FAILURE",
         reason: readableReport(reason)
       })
-    )
-    .chain(({ fiscalCode, choice }) =>
-      userDataProcessingModel
-        .findLastVersionByModelId([
+    ),
+    TE.fromEither,
+    TE.chainW(({ fiscalCode, choice }) =>
+      pipe(
+        userDataProcessingModel.findLastVersionByModelId([
           makeUserDataProcessingId(choice, fiscalCode),
           fiscalCode
-        ])
-        .foldTaskEither<ActivityResultFailure, UserDataProcessing>(
-          error =>
-            fromLeft(
-              ActivityResultQueryFailure.encode({
-                kind: "QUERY_FAILURE",
-                query: "findOneUserDataProcessingById",
-                reason: `${error.kind}, ${getMessageFromCosmosErrors(error)}`
+        ]),
+        TE.mapLeft(error =>
+          ActivityResultQueryFailure.encode({
+            kind: "QUERY_FAILURE",
+            query: "findOneUserDataProcessingById",
+            reason: `${error.kind}, ${getMessageFromCosmosErrors(error)}`
+          })
+        ),
+        TE.chainW(
+          flow(
+            E.fromOption(() =>
+              ActivityResultNotFoundFailure.encode({
+                kind: "NOT_FOUND_FAILURE"
               })
             ),
-          maybeRecord =>
-            maybeRecord.fold(
-              fromLeft(
-                ActivityResultNotFoundFailure.encode({
-                  kind: "NOT_FOUND_FAILURE"
-                })
-              ),
-              _ => taskEither.of(_)
-            )
+            TE.fromEither
```

Comment:
ditto

Reply:
fixed in https://github.com/pagopa/io-functions-admin/pull/165/commits/18d69f92de2dfad63d788ce72a518051528e5c85